packages/fetchai/connections/p2p_noise/noise_node.go

```diff
@@ -62,31 +64,35 @@ func main() {
 	}
 
 	overlay := kademlia.New(kademlia.WithProtocolEvents(events))
+	fmt.Printf("[noise-p2p][info] successfully created overlay!\n")
 
 	// Bind Kademlia to the node.
 	node.Bind(overlay.Protocol())
-
-	// Have the node start listening for new peers.
-	check(node.Listen())
-
-	//
 	fmt.Printf("[noise-p2p][info] started node %s (%s).\n", node.ID().Address, node.ID().ID.String())
 
+	// Once overlay setup, connect to agent
+	check(agent.Connect())
+	fmt.Printf("[noise-p2p][info] successfully connected to AEA!")
+
 	// Ping entry node to initially bootstrap, if non genesis
 	if len(agent.EntryUris()) > 0 {
 		check(bootstrap(node, agent.EntryUris()...))
 	}
 
 	// Attempt to discover peers if we are bootstrapped to any nodes.
-	go func() {
-		for {
-			discover(overlay)
-			time.Sleep(2500 * time.Millisecond)
-		}
-	}()
+	if len(agent.EntryUris()) > 0 {
+		go func() {
+			for {
+				discover(overlay)
+				time.Sleep(2500 * time.Millisecond)
+			}
+		}()
+	}
 
-	// Once overlay setup, connect to agent
-	check(agent.Connect())
+	// Have the node start listening for new peers.
+	// if len(agent.EntryUris()) > 0 {
+	check(node.Listen())
+	// }
```

Comment:
@marcofavorito  and @lrahmani this code is still not sound!
things to note:

- `check(node.Listen())` blocks